src/Exceptions.hpp

```diff
@@ -15,4 +15,8 @@ class ErrnoException final : public std::exception
 private:
     const int _errnum;
     const std::string _message;
-};
\ No newline at end of file
+};
+
+class ClientDisconnectedException final : public std::exception
```

Comment:
As said before, inheriting from base std::exception doesn't give us any added benefits

Reply:
If Ive done that Once I kinda need to do it everywhere, no? I want to catch std::exception and have this caught as well…